### DISCUSSION TITLE
PUT Requests with multipart/form-data Fail to Include File Data

### DISCUSSION THREAD

**Discussion by yuuri111** [RobinTail/express-zod-api#1698] [Q&A] [answered]
Hello,

I am encountering a problem with `PUT` requests for file uploads using `multipart/form-data` in my project. While `POST` requests with the same file data are processed successfully, `PUT` requests fail to include the file in the request payload.

## Issue Description
When I send a `POST` request with `multipart/form-data` including a file, the server-side receives the `UploadedFile` object as expected and processes the file upload successfully. However, the same process with a `PUT` request results in an `undefined` file object on the server-side.

## Steps to Reproduce
1. Send a `POST` request with `multipart/form-data` including a file, which succeeds.
2. Send a `PUT` request with `multipart/form-data` including the same file, which fails.

## Expected Behavior
The `PUT` request should handle the file upload in the same manner as the `POST` request, and the `UploadedFile` object should be received correctly on the server-side.

## Actual Behavior
When sending a `PUT` request, the server-side does not receive the `UploadedFile` object, and the request body has an `undefined` file object. The following error message is shown:

I would greatly appreciate any assistance or insight into this issue. Thank you in advance for your time and help.

Best regards,

- config
```
export const config = createConfig({
  server: {
    listen: 3000, 
    upload: true
  },
  cors: true,
  logger: { level: "debug", color: true },
});
```

- API code
```
upload: defaultEndpointsFactory.build({
    method: "put",
    tag: "upload",
    description: "Upload a file.",
    input: z.object({
      file: ez.upload(),
    }),
    output: z.object({}),
    handler: async ({ input }) => {
      console.log(input);
      return { url: input.file };
    },
  }),
```
- request & response
<img width="1470" alt="スクリーンショット 2024-04-24 20 48 36" src="https://github.com/RobinTail/express-zod-api/assets/9194205/2040a262-e096-45f4-95ec-1fa363216a1c">

**Comment by RobinTail** [comment, marked as answer]
Hello @yuuri111 ,

You need to add the `inputSources` to your config, because the inputs aggregation from `files` is only configured for `POST` by default. So you can adjust it for your needs this way:

```ts
export const config = createConfig({
  inputSources: {
    put: ["body", "params", "files"], // <———
  },
  server: {
    listen: 3000, 
    upload: true
  },
  cors: true,
  logger: { level: "debug", color: true },
});
```

Details: https://github.com/RobinTail/express-zod-api#customizing-input-sources

Let me know if it worked for you.

**Comment by yuuri111** [comment]
@RobinTail 
This works great! Thank you.
